### ISSUE TITLE
unit testing for all the documented test case

### ISSUE THREAD

**Issue by akshitababel** [KIPFoundation/kfs#146]

**Comment by 0zAND1z** [comment]
Hi @akshitababel , please check the updated list of modules that needs to be tested under this User Story. 

Also, let me know if we have missed any module in this list. Thanks.

**Comment by 0zAND1z** [comment]
If the list is good & final, let's have separate task cards created for each one of these.

**Comment by akshitababel** [comment]
@0zAND1z  matchsafe, diauth, common, cluster, chachedata are the packages internally used by the function in file, folder, remove and readdata and they are getting tested everytime when all these four packages are tested, and we can't test those packages separately. The test cases will be written only for file, folder, remove and readdata  and this test cases are internally dependent on rest of the packages so those will also get tested when we will run the test cases for these 4 package.

**Comment by 0zAND1z** [comment]
So, are if the test functions fail in one of your packages, will it note/point to the respective packages that are involved?

For example - You're testing a file upload test case & the Handler fails to return SUCCESS. Let's assume the reason is that the .pub keys are not stored/properly copied into respective folder. What then? Please let me know.

**Comment by akshitababel** [comment]
When the test case fails it will just give the status code of why it failed but it will not point out the specific package due to which it failed

**Comment by 0zAND1z** [comment]
Right, so we can not know if the test failed because of the environment issue or if its a package failure.

That's exactly why I am proposing to have test for those internal packages as well.
That way, we get to a granular level of understanding why a function failed. Isn't it a good idea to have unit tests for the dependency packages so that we can at-the-least, rule them out as point of failure?

Let me know if that makes sense :)

**Comment by 0zAND1z** [comment]
Hi @akshitababel - Post our call, here is what i think we need to do for `matchsafe`

1. Modify the Initenc() to ensure that the function returns 2 values - the SecureEnclave and the error(nil if its a success)
2. Write a unit test to assert the above run-time behaviour.

I will help you here.. Just a ping away.

**Comment by akshitababel** [comment]
@0zAND1z  Ok, I will proceed like this
